src/mpl/src/str/mpl_argstr.c

```diff
@@ -71,7 +71,8 @@ static int decode_buffer(const char *str, char *dest, int length, int *num_decod
         str++;
         hex[1] = *str;
         str++;
-        sscanf(hex, "%X", &value);
+        if (0 == sscanf(hex, "%X", &value))
+            return MPL_STR_TRUNCATED;
```

Comment:
Is truncation the right error to return here? 

> On success, these functions return the number of input items successfully matched and assigned; this can be fewer than provided for, or even zero, in the event of an early matching failure.

Reply:
It's either that or `MPL_STR_FAIL`. I do think getting `0` here is an error case though, unless I'm not understanding the logic correctly.